### PR TITLE
Consolidate group and instance selection code

### PR DESCRIPTION
**Pull request by jemoreira**
It was spread between some commands, the selectors in cli and the instance manager.

### CODE REVIEW

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h [hunk 1]
```diff
@@ -16,12 +16,28 @@
...
+#include "host/commands/cvd/instances/instance_record.h"
 
 namespace cuttlefish {
-Result<LocalInstanceGroup> SelectGroup(InstanceManager&, const CommandRequest&);
+namespace selector {
+
+// Selects a single group based on the request's selector options. Asks the user
+// to manually choose a single group if multiple groups match the selector
+// options and stdin is a terminal.
+Result<LocalInstanceGroup> SelectGroup(const InstanceManager& instance_maanger,
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h:33]: nit: typo

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h [hunk 2]
```diff
@@ -16,12 +16,28 @@
...
+                                       const CommandRequest& request);
+
+// Selects a single instance based on the reqeust's selector options. Unlike
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h:36]: nit: typo

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h [hunk 3]
```diff
@@ -16,12 +16,28 @@
...
+// SelectGroup it doesn't ask the user to refine the selection in case multiple
+// instances match. Also returns the group the selected instance belongs to.
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h:38]: if multiple instances match, how do we select one then?

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp [hunk 4]
```diff
@@ -93,17 +128,38 @@ Result<LocalInstanceGroup> PromptUserForGroup(InstanceManager& instance_manager,
   }
 }
 
+Result<LocalInstanceGroup> FindGroupOrDefault(
+    const InstanceDatabase::Filter& filter,
+    const InstanceManager& instance_manager) {
+  if (filter.Empty()) {
+    auto default_group = CF_EXPECT(GetDefaultGroup(instance_manager));
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp:135]: nit: can these two lines be merged into one: `return CF_EXPECT(GetDefaultGroup(instance_manager));`

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp [hunk 5]
```diff
@@ -93,17 +128,38 @@ Result<LocalInstanceGroup> PromptUserForGroup(InstanceManager& instance_manager,
...
+    return default_group;
+  }
+  auto groups = CF_EXPECT(instance_manager.FindGroups(filter));
+  CF_EXPECT(groups.size() == 1, "groups.size() = " << groups.size());
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp:139]: should we use ` CF_EXPECT_EQ` to be consistent as in  line 147 with `CF_EXPECT_EQ(instances.size(), 1u, ...`?

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp [hunk 6]
```diff
@@ -114,4 +170,17 @@ Result<LocalInstanceGroup> SelectGroup(InstanceManager& instance_manager,
       PromptUserForGroup(instance_manager, request, env, std::move(filter)));
 }
 
+Result<std::pair<LocalInstance, LocalInstanceGroup>> SelectInstance(
+    const InstanceManager& instance_manager, const CommandRequest& request) {
+  InstanceDatabase::Filter filter =
+      CF_EXPECT(BuildFilterFromSelectors(request.Selectors(), request.Env()));
+
+  if (filter.Empty()) {
```
Comment (ser-io) [inline base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp:178]: nit: use ternary conditional operator: `return filter.Empty() ? ...  : ...`?

**Review by ser-io — APPROVED** [review]

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp (hunk 4 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp (hunk 5 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp (hunk 6 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.cpp]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h (hunk 1 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h (hunk 2 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h]: Done

File: base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h (hunk 3 above)
Comment (jemoreira) [reply base/cvd/cuttlefish/host/commands/cvd/cli/selector/selector.h]: It fails, added to the comment.

**Pull request merged**